geo_map/api/views.py

```diff
@@ -62,3 +70,43 @@ def get_queryset(self):
         qs = qs.exclude(provider__isnull=True)
 
         return qs
+
+
+class ProviderViewSet(ModelViewSet):
+    queryset = Provider.objects.all()
+    serializer_class = ProviderSerializer
+
+    def get_queryset(self):
+        return Provider.objects.all()
+
+    def list(self, request, *args, **kwargs):
+        cached_providers = cache.get("cached_providers")
+        if cached_providers is None:
+            queryset = self.get_queryset()
+            serialized_providers = ProviderSerializer(queryset, many=True).data
+            cached_providers = serialized_providers
+            cache.set(
+                "cached_providers", cached_providers, 1800
+            )  # Cache for 30 minutes
+        return Response(cached_providers)
+
+
+class ProviderListAPIView(APIView):
+    def get(self):
+        providers = (
+            Provider.objects.annotate(circuit_count=Count("circuits"))
+            .prefetch_related("circuits__terminations__site")
+            .filter(circuit_count__gt=0)
+            .order_by("name")
+        )
+
+        providers_data = [
+            {
+                "value": provider.id,
+                "label": provider.name,
+                "color": provider.cf.get("provider_color"),
+            }
+            for provider in providers
+        ]
+
+        return Response(providers_data)
```

Comment:
1. This is uncached
2. Why not use the `ProviderViewSet`? You're overriding the route handler in `geo_map/urls.py`